### PR TITLE
IGNITE-11723: IgniteSpark integration should support skipStore option for internal dataStreamer (IgniteRdd and Ignite DataFrame)

### PR DESCRIPTION
**Pull request by zaleslaw**

### CODE REVIEW

**Comment by zaleslaw** [comment]
Dear @nizhikov this is my first commit to the Spark component, please have a look and approve this change

**Comment by nizhikov** [comment]
@zaleslaw Thanks for the PR.

I will do review, shortly.

**Comment by zaleslaw** [comment]
@nizhikov Also, I've run on TC next build: RunALL, does it includes Spark component and exammples or I should run something else? 
Please, have a look during the review https://ci.ignite.apache.org/viewLog.html?buildId=4633586&buildTypeId=IgniteTests24Java8_RunAll

File: modules/spark/src/main/scala/org/apache/ignite/spark/IgniteRDD.scala [hunk 1]
```diff
@@ -223,13 +222,14 @@ class IgniteRDD[K, V] (
...
 
     /**
-     * Saves values from the given key-value RDD into Ignite.
-     *
-     * @param rdd RDD instance to save values from.
-     * @param overwrite Boolean flag indicating whether the call on this method should overwrite existing
-     *      values in Ignite cache.
-     */
-    def savePairs(rdd: RDD[(K, V)], overwrite: Boolean = false) = {
+      * Saves values from the given key-value RDD into Ignite.
```
Comment (nizhikov) [inline modules/spark/src/main/scala/org/apache/ignite/spark/IgniteRDD.scala]: Seems you added space for existing comment.
Please, remove it.

Comment (nizhikov) [inline modules/spark/src/main/scala/org/apache/ignite/spark/IgniteRDD.scala]: Why thins reformatting?
Please, revert unnecessary format changes.

File: modules/spark/src/main/scala/org/apache/ignite/spark/IgniteRDD.scala [hunk 2]
```diff
@@ -254,12 +255,13 @@ class IgniteRDD[K, V] (
     /**
      * Saves values from the given RDD into Ignite.
      *
-     * @param rdd RDD instance to save values from.
-     * @param f Transformation function.
-     * @param overwrite Boolean flag indicating whether the call on this method should overwrite existing
-     *      values in Ignite cache.
+     * @param rdd        RDD instance to save values from.
```
Comment (nizhikov) [inline modules/spark/src/main/scala/org/apache/ignite/spark/IgniteRDD.scala]: Why thins reformatting?
Please, revert unnecessary format changes.

File: modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala [hunk 3]
```diff
@@ -82,20 +82,21 @@ class JavaIgniteRDD[K, V](override val rdd: IgniteRDD[K, V])
...
-        rdd.savePairs(rrdd, overwrite)
+        rdd.savePairs(rrdd, overwrite, skipStore)
     }
 
-    def savePairs(jrdd: JavaPairRDD[K, V]) : Unit = savePairs(jrdd, overwrite = false)
+    def savePairs(jrdd: JavaPairRDD[K, V]): Unit = savePairs(jrdd, overwrite = false, skipStore = false)
 
-    def savePairs[T](jrdd: JavaRDD[T], f: (T, IgniteContext) ⇒ (K, V), overwrite: Boolean = false) = {
-        rdd.savePairs(JavaRDD.toRDD(jrdd), f, overwrite)
+    def savePairs[T](jrdd: JavaRDD[T], f: (T, IgniteContext) ⇒ (K, V), overwrite: Boolean = false,
```
Comment (nizhikov) [inline modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala]: Redundant NL

**Comment by nizhikov** [comment]
@zaleslaw 

Overall looks good.

1. Seems your IDE code format changes differ from Ignite one. There is some nits in patch. Please, revert unnecessary changes.

2. Can we add the test for a new property?

**Comment by zaleslaw** [comment]
About formatting, I use codestyle for two years the same like for other
components, ok, I'll revert the formatting, but could you share please the
link or file with codestyle (maybe somebody used outdated) and this is an
issue.

About test - I thought, it's not to easy to emulate because but I'll try

ср, 25 сент. 2019 г. в 22:59, Nikolay <notifications@github.com>:

File: modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala (hunk 3 above)
Comment (zaleslaw) [reply modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala]: I'm sorry, redundant what? What is NL in your comment above?

**Comment by nizhikov** [comment]
https://cwiki.apache.org/confluence/display/IGNITE/Coding+Guidelines - Code formatting guide.

File: modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala (hunk 3 above)
Comment (nizhikov) [reply modules/spark/src/main/scala/org/apache/ignite/spark/JavaIgniteRDD.scala]: Seems, these two lines can be combined in one.
I double-checked and seems it not.

Sorry, just skip this comment.

**Comment by zaleslaw** [comment]
@nizhikov Could you please revisit your review (I've improved formatting and added special test-case for the skipStore enabling behaviour

File: modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java [hunk 4]
```diff
@@ -150,6 +169,59 @@ public void testStoreDataToIgnite() throws Exception {
...
+
+            // 1000 entries in store; 1000 entries in cache
+            for (int i = 0; i < 1000; i++)
+                storeMap.put(i, i);
+
+            // 1000 entries in store and cache; 1000 another entries in cache writing to store
+            ic.fromCache(PARTITIONED_CACHE_NAME)
+                .savePairs(sc.parallelize(F.range(1000, 2000), GRID_CNT).mapToPair(SIMPLE_FUNCTION), true, false);
+
+            // check 2000 entries in store
```
Comment (nizhikov) [inline modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java]: Nit: The sentence should start with the capital letter and ends with the dot.

File: modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java [hunk 5]
```diff
@@ -150,6 +169,59 @@ public void testStoreDataToIgnite() throws Exception {
...
+
+            Ignite ignite = ic.ignite();
+
+            IgniteCache<Integer, Integer> cache = ignite.cache(PARTITIONED_CACHE_NAME);
+
+            // 1000 entries in store; 1000 entries in cache
+            for (int i = 0; i < 1000; i++)
+                storeMap.put(i, i);
+
+            // 1000 entries in store and cache; 1000 another entries in cache writing to store
```
Comment (nizhikov) [inline modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java]: Nit: The sentence should start with the capital letter and ends with the dot.

File: modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java [hunk 6]
```diff
@@ -150,6 +169,59 @@ public void testStoreDataToIgnite() throws Exception {
...
+        JavaIgniteContext<Integer, Integer> ic = null;
+
+        try {
+            ic = new JavaIgniteContext<>(sc, new IgniteConfigProvider(), false);
+
+            Ignite ignite = ic.ignite();
+
+            IgniteCache<Integer, Integer> cache = ignite.cache(PARTITIONED_CACHE_NAME);
+
+            // 1000 entries in store; 1000 entries in cache
```
Comment (nizhikov) [inline modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java]: Nit: The sentence should start with the capital letter and ends with the dot.

File: modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java [hunk 7]
```diff
@@ -150,6 +169,59 @@ public void testStoreDataToIgnite() throws Exception {
...
+
+            // 1000 entries in store and cache; 1000 another entries in cache writing to store
+            ic.fromCache(PARTITIONED_CACHE_NAME)
+                .savePairs(sc.parallelize(F.range(1000, 2000), GRID_CNT).mapToPair(SIMPLE_FUNCTION), true, false);
+
+            // check 2000 entries in store
+            for (int i = 0; i < 2000; i++)
+                assertEquals(i, storeMap.get(i));
+
+            // add 1000 entries to cache but skip store
```
Comment (nizhikov) [inline modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java]: Nit: The sentence should start with the capital letter and ends with the dot.

File: modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java [hunk 8]
```diff
@@ -335,4 +412,25 @@ private static IgniteConfiguration getConfiguration(String igniteInstanceName, b
             return t._2();
         }
     }
+
+    /**
+     *
```
Comment (nizhikov) [inline modules/spark/src/test/java/org/apache/ignite/spark/JavaEmbeddedIgniteRDDSelfTest.java]: Nit: We should write a javadoc here or replace it with the 1 line stub `/** */`

**Comment by nizhikov** [comment]
@zaleslaw 

LGTM, except for a couple of nits I mentioned in comments.
We should run at least `Spark RDD` and `Examples` suites to test this patch.

Please, check suites still succeed before the merge.

**Comment by zaleslaw** [comment]
@nizhikov Thanks a lot, will run all required suites

**Pull request merged**